extension/src/codeLensProvider.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { CancellationToken, CodeLens, CodeLensProvider, EventEmitter, TextDocument } from 'vscode';
+import { JavaTestRunnerCommands } from './constants/commands';
+import { ITestItem } from './protocols';
+import { searchTestCodeLens } from './utils/commandUtils';
+
+class TestCodeLensProvider implements CodeLensProvider {
+    private _onDidChangeCodeLenses: EventEmitter<void> = new EventEmitter<void>();
+
+    public refresh(): void {
```

Comment:
who will call it?

Reply:
For now, no one will call it.

It might be used in the future, when, for example, tests run finished, and we need to tell the code lens provider to update the status in the code lens.